File: algorithms/algorithms-cplusplus/leetcode/Question_0455_Assign_Cookies.cc
#include <vector>
#include <algorithm>

using namespace std;

class Solution {
 public:
  int findContentChildren(vector<int>& g, vector<int>& s) {
    if (g.empty() || s.empty()) {
      return 0;
    }
    sort(g.begin(), g.end());
    sort(s.begin(), s.end());
    int res = 0;
    for (int i = 0, j = 0; j < s.size(); j++) {
      if (s[j] >= g[i]) {
        res++;
        i++;
        if (i >= g.size()) {
          break;
        }
      }
    }
    return res;
  }
};

File: algorithms/algorithms-java/src/main/java/com/lanceolata/coding/algorithms/java/leetcode/Question_0455_Assign_Cookies.java
package com.lanceolata.coding.algorithms.java.leetcode;

import java.util.Arrays;

public class Question_0455_Assign_Cookies {
    public int findContentChildren(int[] g, int[] s) {
        if (g == null || s == null) {
            return 0;
        }
        int res = 0;
        Arrays.sort(g);
        Arrays.sort(s);
        for (int i = 0, j = 0; j < s.length; j++) {
            if (s[j] >= g[i]) {
                res++;
                i++;
                if (i >= g.length) {
                    break;
                }
            }
        }
        return res;
    }
}
